Wikipedia/Code/WMFExploreSectionControllerCache.m

```diff
@@ -43,22 +50,80 @@ - (instancetype)initWithDataStore:(MWKDataStore*)dataStore {
         self.dataStore                              = dataStore;
         self.sectionControllersBySection            = [[NSCache alloc] init];
         self.sectionControllersBySection.countLimit = [WMFExploreSection totalMaxNumberOfSections];
-        self.sectionsBySectionController            = [NSMapTable mapTableWithKeyOptions:NSMapTableWeakMemory | NSMapTableObjectPointerPersonality
-                                                                            valueOptions:NSMapTableWeakMemory];
+
+        /*
+         Controllers (keys in the map) are retained by the NSCache and can be evicted at any time. On the other hand,
+         sections (values in the map) are "value types" which should be owned by the receiver.  In other words, sections
+         are created as needed, and not retained for any particular reason, and it would be unintentional for a value in
+         this map to disappear because it was dereferenced externally.
+         */
+        self.sectionsBySectionController            =
+            [NSMapTable mapTableWithKeyOptions:NSMapTableWeakMemory | NSMapTableObjectPointerPersonality
+                                  valueOptions:NSMapTableStrongMemory];
     }
     return self;
 }
 
+#if DEBUG
```

Comment:
sort of, this removes the actual implementation. the macro above ensures it only tries to call it in debug configurations. otherwise the methods would remain even though they're never called.